dependencies/http/http.go

```diff
@@ -142,3 +142,28 @@ func WithTLSConfig(c Client, config *tls.Config) (Client, error) {
 	}
 	return &newClient, nil
 }
+
+// PrivateClient is an http client that obscures error messages that may contain
+// sensitive information
+type PrivateClient struct {
```

Comment:
Could you make this type unexported (lowercase letter) since you have a public constructor that returns an interface below?

Is it possible to write a unit test that shows that it always returns the opaque error?

Reply:
Had to make this public again because there's a test in a separate package that references this type.